### PR TITLE
Docker support on EMR

### LINKED ISSUE #2179: support docker on EMR 6.x AMIs
EMR [now has explicit support for Docker on 6.x AMIs](https://aws.amazon.com/blogs/big-data/run-spark-applications-with-docker-using-amazon-emr-6-0-0-beta/). mrjob should integrate with this

### PR DESCRIPTION
**Pull request by davidmarin**
Adds support for running Spark executors in Docker containers on EMR. It's activated by setting the `docker_image` option. Also added the `docker_mounts` option, and the `docker_client_config` option (so you can log into ECR). Fixes #2179.

If you want to test this out without building a Docker image, try `python -m mrjob.examples.mr_spark_wordcount --docker-image amazoncorretto:8` on Python 2.

### CODE REVIEW

File: mrjob/cloud.py [hunk 1]
```diff
@@ -77,9 +77,6 @@ class HadoopInTheCloudJobRunner(MRJobBinRunner):
         'zone',
     }
 
-    # so far, every service provides the ability to run bootstrap scripts
-    _BOOTSTRAP_MRJOB_IN_PY_FILES = False
```
Comment (88manpreet) [inline mrjob/cloud.py:81]: Why is this removed?

File: mrjob/emr.py [hunk 2]
```diff
@@ -712,17 +717,10 @@ def _check_output_not_exists(self):
     def _add_bootstrap_files_for_upload(self, persistent=False):
         """Add files needed by the bootstrap script to self._upload_mgr.
 
-        Tar up mrjob if bootstrap_mrjob is True.
-
         Create the master bootstrap script if necessary.
 
         persistent -- set by make_persistent_cluster()
         """
-        # lazily create mrjob.zip
```
Comment (88manpreet) [inline mrjob/emr.py:721]: Why are we removing this?

**Comment by 88manpreet** [comment]
Looks good to me. Left some questions. +1 for merging this.

File: mrjob/cloud.py (hunk 1 above)
Comment (davidmarin) [reply mrjob/cloud.py:81]: Because installing mrjob on the cluster at bootstrap time no longer helps when we're using Docker, since that won't impact what libraries are available inside the Docker container.

File: mrjob/emr.py (hunk 2 above)
Comment (davidmarin) [reply mrjob/emr.py:721]: Because mrjob is no longer part of bootstrapping; it's now part of `py_files` like on the Hadoop runner.
